### PR TITLE
[Jenkins] Bump cppcheck to 2.0

### PR DESCRIPTION
**Pull request by bilke**

### CODE REVIEW

**Comment by endJunction** [comment]
Looks like it [needs](https://jenkins.opengeosys.org/blue/organizations/jenkins/ufz%2Fogs/detail/PR-2966/1/pipeline) clang-tidy:
[2020-05-26T11:34:28.358Z] -- CLANG_TIDY=CLANG_TIDY-NOTFOUND
[2020-05-26T11:34:28.358Z] -- Configuring incomplete, errors occurred!

**Comment by bilke** [comment]
@endJunction Here are [the results](https://jenkins.opengeosys.org/job/ufz/job/ogs/job/PR-2966/2/cppcheck/).

**Comment by bilke** [comment]
Shall we set a limit for allowed warnings? We did not had a limit yet.

**Comment by endJunction** [comment]
Let's keep it limitless. Will set it after fixing few hundreds.

**Comment by bilke** [comment]
Ups I just set it... You can lower it later on!

File: scripts/docker/Dockerfile.gcc.gui [hunk 1]
```diff
@@ -114,14 +114,14 @@ RUN mkdir -p /var/tmp && wget -q -nc --no-check-certificate -P /var/tmp https://
...
 
-# cppcheck version 1.87
-# https://github.com/danmar/cppcheck/archive/1.87.tar.gz
-RUN mkdir -p /var/tmp && wget -q -nc --no-check-certificate -P /var/tmp https://github.com/danmar/cppcheck/archive/1.87.tar.gz && \
-    mkdir -p /var/tmp && tar -x -f /var/tmp/1.87.tar.gz -C /var/tmp -z && \
-    mkdir -p /var/tmp/cppcheck-1.87/build && cd /var/tmp/cppcheck-1.87/build && cmake -DCMAKE_INSTALL_PREFIX=/usr/local/cppcheck -D CMAKE_BUILD_TYPE=Release /var/tmp/cppcheck-1.87 && \
-    cmake --build /var/tmp/cppcheck-1.87/build --target all -- -j$(nproc) && \
-    cmake --build /var/tmp/cppcheck-1.87/build --target install -- -j$(nproc) && \
-    rm -rf /var/tmp/cppcheck-1.87 /var/tmp/1.87.tar.gz
+# cppcheck version 809a769c690d8ab6fef293e41a29c8490512866e
```
Comment (TomFischer) [inline scripts/docker/Dockerfile.gcc.gui:117]: Version number is not very easy to remember ;-)

File: Jenkinsfile [hunk 2]
```diff
@@ -227,7 +227,8 @@ pipeline {
                              pattern: 'build/build*.log')],
                 unstableTotalAll: 1
               recordIssues enabledForFailure: true,
-                tools: [cppCheck(pattern: 'build/cppcheck.log')]
+                tools: [cppCheck(pattern: 'build/cppcheck.log')],
+                unstableTotalAll: 405
```
Comment (endJunction) [inline Jenkinsfile:231]:
```suggestion
                unstableTotalAll: 500
```

File: scripts/docker/Dockerfile.gcc.gui (hunk 1 above)
Comment (bilke) [reply scripts/docker/Dockerfile.gcc.gui:117]: It is just a specific commit as version 2.0 had a bug in its CMake config. The cppcheck people are working on the 2.1 release which we will then use...

**Pull request merged**